transport/http/handler.go

```diff
@@ -227,14 +234,20 @@ func (rw *responseWriter) AddHeaders(h transport.Headers) {
 }
 
 func (rw *responseWriter) SetApplicationError() {
-	rw.w.Header().Set(ApplicationStatusHeader, ApplicationErrorStatus)
+	rw.isApplicationError = true
+}
+
+func (rw *responseWriter) UpdateFeatures(f func(*transport.ResponseFeatures)) {
+	f(&rw.features)
 }
 
 func (rw *responseWriter) AddSystemHeader(key string, value string) {
 	rw.w.Header().Set(key, value)
 }
 
 func (rw *responseWriter) Close(httpStatusCode int) {
+	rw.w.Header().Set(ApplicationStatusHeader, applicationStatusValue(rw.isApplicationError))
+	rw.w.Header().Set(AcceptResponseErrorHeader, acceptValue(rw.features.AcceptResponseError))
```

Comment:
The response header should not be `Accept-*`, but rather just `*`, to indicate the server’s election to support the feature, and absent if the server does not recognize the feature request.

Reply:
Updated